virttest/libvirt_xml/vm_xml.py

```diff
@@ -1994,8 +1994,231 @@ def marshal_to_timer(tag, attr_dict, index, libvirtxml):
+        if element is not None:
+            xmltreefile.remove(element)
+            xmltreefile.write()
+
+    # Sub-element of CacheTuneXML
+    class MonitorXML(VMXML):
+
+        """Monitor element of CacheTuneXML"""
+
+        __slots__ = ('level', 'vcpus')
+
+        def __init__(self, virsh_instance=base.virsh):
+            """
+            Create new MonitorXML instance
+            """
+            accessors.XMLAttribute(property_name='level',
+                                   libvirtxml=self,
+                                   forbidden=[],
+                                   parent_xpath='/',
+                                   tag_name='monitor',
+                                   attribute='level')
+            accessors.XMLAttribute(property_name="vcpus",
+                                   libvirtxml=self,
+                                   forbidden=[],
+                                   parent_xpath='/',
+                                   tag_name='monitor',
+                                   attribute='vcpus')
+
+            super(CacheTuneXML.MonitorXML, self).__init__(
+                virsh_instance=virsh_instance)
+            self.xml = '<monitor/>'
+
+    def get_monitor(self):
+        xmltreefile = self.__dict_get__('xml')
+        try:
+            monitor_root = xmltreefile.reroot('/monitor')
+        except KeyError as detail:
+            raise xcepts.LibvirtXMLError(detail)
+        monitorxml = CacheTuneXML.MonitorXML(virsh_instance=self.__dict_get__('virsh'))
+        monitorxml.xmltreefile = monitor_root
+        return monitorxml
+
+    def set_monitor(self, value):
+        if not issubclass(type(value), CacheTuneXML.MonitorXML):
+            raise xcepts.LibvirtXMLError("value must be a Monitor or subclass")
+        xmltreefile = self.__dict_get__('xml')
+        # MonitorXML root element is whole Monitor element tree
+        root = xmltreefile.getroot()
+        root.append(value.xmltreefile.getroot())
+        xmltreefile.write()
+
+    def del_monitor(self):
+        xmltreefile = self.__dict_get__('xml')
+        element = xmltreefile.find('/monitor')
+        if element is not None:
+            xmltreefile.remove(element)
+            xmltreefile.write()
+
+
+class MemoryTuneXML(VMXML):
+
+    """Event element of perf"""
+
+    __slots__ = ('vcpus', 'node')
+
+    def __init__(self, virsh_instance=base.virsh):
+        """
+        Create new MemoryTuneXML instance
+        """
+        accessors.XMLAttribute(property_name="vcpus",
+                               libvirtxml=self,
+                               forbidden=[],
+                               parent_xpath='/',
+                               tag_name='memorytune',
+                               attribute='vcpus')
+
+        super(MemoryTuneXML, self).__init__(
+            virsh_instance=virsh_instance)
+        self.xml = '<memorytune/>'
+
+    # Sub-element of MemoryTuneXML
+    class NodeXML(VMXML):
```

Comment:
Can we define it out of scope MemoryTuneXML, rather than inner class?

Reply:
The <node> is nested in the <memorytune>  in the xml. So the script follows the same structure and make it consistent and clear.